storage/src/lsm_tree/components/block.rs

```diff
@@ -43,21 +43,21 @@ impl Block {
                     .read_to_end(&mut decoded)
                     .map_err(Error::decode_error)
                     .unwrap();
-                Bytes::from(decoded)
+                decoded
             }
         };
 
         // Decode restart points.
         let n_restarts = (&buf[buf.len() - 4..]).get_u32_le();
         let data_len = buf.len() - 4 - n_restarts as usize * 4;
         let mut restart_points = Vec::with_capacity(n_restarts as usize);
-        let mut restart_points_buf = &buf[data_len..buf.len() - 4];
+        let mut restart_points_buf = &buf[data_len..buf.len() - 4]; //fixme: remove mut
         for _ in 0..n_restarts {
             restart_points.push(restart_points_buf.get_u32_le());
         }
 
         Ok(Block {
-            data: buf.slice(..data_len),
+            data: buf[..data_len].to_vec(), // clone
```

Comment:
No need to copy the buffer for it will not be used anywhere. Please truncate the buffer and directly move it.